pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -70,3 +86,100 @@ func (r *reconciler) Reconcile(ctx context.Context) error {
 
 	return r.createOrUpdateStatefulset(ctx)
 }
+
+func (r *reconciler) reconcileTlsSecret(ctx context.Context) error {
+	query := k8ssecret.Query(r.client, r.client, log)
+
+	if r.dk.GetExtensionsTlsRefName() != "" {
+		return query.DeleteForNamespaces(ctx, getSelfSignedTlsSecretName(r.dk.Name), []string{r.dk.Namespace})
```

Comment:
Why not use just `return query.Delete(...)` ? As long as the secret has a correct metadata section it will work

Reply:
[Fixed](https://github.com/Dynatrace/dynatrace-operator/pull/3778/commits/effefb722de190fad5f3e2e58bdde6121b9e64dc)